softwareComponents/simplesimClientLib/src/changecolor.cpp

```diff
@@ -0,0 +1,97 @@
+#include "changecolor.hpp"
+#include "ui_changecolor.h"
+
+#include <iostream>
+
+using rofi::simplesim::ChangeColor;
+
+ChangeColor::ChangeColor( QWidget* parent, size_t size ) :
+    ui( new Ui::ChangeColor ),
+    parent( parent ),
+    to_color( size, false )
+{
+    //QWidget( parent );
+    ui->setupUi( this );
+    ui->plainTextEdit->hide();
+
+    connect( ui->buttonBox, SIGNAL( rejected() ), this, SLOT( hide() ) );
+    connect( ui->buttonBox, SIGNAL( accepted() ), this, SLOT( accept() ) );
+    connect( ui->toolButton, SIGNAL( clicked() ), this, SLOT( showHelp() ) );
+}
+
+ChangeColor::~ChangeColor(){
+    delete ui;
+}
+
+void ChangeColor::accept(){
+    hide();
+    try {
+        parseRange();
+    } catch ( std::exception& e ){
```

Comment:
When you have a custom exception type, you can catch it more precisely. This is more error-prone.